src/main/java/org/opensearch/flowframework/common/FlowFrameworkSettings.java

```diff
@@ -133,22 +139,49 @@ public class FlowFrameworkSettings {
         Setting.Property.NodeScope
     );
 
+    /** This setting sets max retryable tasks during workflow execution that polls results */
+    public static final Setting<Integer> WORKFLOW_THREAD_POOL_SIZE = Setting.intSetting(
```

Comment:
Do we want to keep this setting as dynamic or final?

Reply:
It should not be dynamic, as it is used in Thread Pool initialization and it is not possible to resize thread pools once started.

It is currently static (there is no `Dynamic` property), where it can be changed on a node restart.

I do not think it needs to be final (where it would survive a node restart).

Reply:
> We are not making thread pool sizes configurable then :)

We are making them configurable in `opensearch.yml` or command line parameters, which is the intent. :)